components/checkbox/stories/template.js

```diff
@@ -62,13 +62,19 @@ export const Template = ({
 				type="checkbox"
 				class="${rootClass}-input"
 				aria-labelledby=${ifDefined(ariaLabelledby)}
+				aria-disabled=${ifDefined(isReadOnly ? "true" : undefined)}
 				?checked=${isChecked}
-				?disabled=${isDisabled || isReadOnly}
+				?disabled=${isDisabled}
 				title=${ifDefined(title)}
 				value=${ifDefined(value)}
-				@change=${function() {
-					if (isDisabled) return;
-					updateArgs({ isChecked: !isChecked });
+				@change=${(e) => {
+					if (isReadOnly) {
+						// Make checked value immutable for read-only.
+						e.preventDefault();
+						e.target.checked = !e.target.checked;
+					}
+					if (isDisabled || isReadOnly) return;
```

Comment:
Would we ever get to this "or" portion of the code, the `|| isReadOnly`?

The code would run through the first `if(isReadOnly)`, and if it's also disabled, it'll return anyways, but I guess my question is do we need the "or read only" in this `if(isDisabled || isReadOnly)` block. Could it run through _both_ if blocks, if a read-only checkbox isn't disabled? Should the return just go in the `if(isReadOnly)` block?

Reply:
Yes, since readonly is not disabled. This is as intended. If readonly, it keeps the value from changing on click. Then it returns if it's disabled or readonly, and only runs updateArgs to changed the isChecked control value for checkboxes that aren't disabled or readonly.